py/vtdb/vtgate_utils.py

```diff
@@ -75,3 +77,71 @@ def wrapper(self, *args, **kwargs):
           delay = min(max_delay_ms, delay)
     return wrapper
   return decorator
+
+class VitessError(Exception):
+  """VitessError is raised by an RPC  with a server-side application error.
```

Comment:
I made it a child of Exception intentionally. My thinking is that it should only be used internally within the client, and it would be an error to catch it outside our client code. I'd prefer if it crashes and reveals an error, instead of accidentally catching it.